analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarLintXmlReaderTest.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.UnitTest.Helpers;
+
+[TestClass]
+public class SonarLintXmlReaderTest
+{
+    [DataTestMethod]
+    [DataRow(LanguageNames.CSharp, "cs")]
+    [DataRow(LanguageNames.VisualBasic, "vbnet")]
+    public void SonarLintXml_WhenAllValuesAreSet_ExpectedValues(string language, string propertyLanguage)
+    {
+        var sut = CreateSonarLintXmlReader($"ResourceTests\\SonarLintXml\\All_Properties_{propertyLanguage}\\SonarLint.xml", language);
+        sut.Settings.IgnoreHeaderComments.Should().BeTrue();
+        sut.Settings.AnalyzeGeneratedCode.Should().BeFalse();
+        TestArray(sut.Settings.Exclusions, nameof(sut.Settings.Exclusions));
+        TestArray(sut.Settings.Inclusions, nameof(sut.Settings.Inclusions));
+        TestArray(sut.Settings.GlobalExclusions, nameof(sut.Settings.GlobalExclusions));
+        TestArray(sut.Settings.TestExclusions, nameof(sut.Settings.TestExclusions));
+        TestArray(sut.Settings.TestInclusions, nameof(sut.Settings.TestInclusions));
+        TestArray(sut.Settings.GlobalTestExclusions, nameof(sut.Settings.GlobalTestExclusions));
+
+        static void TestArray(string[] array, string folder)
+        {
+            array.Should().HaveCount(2);
+            array[0].Should().BeEquivalentTo($"Fake/{folder}/**/*");
+            array[1].Should().BeEquivalentTo($"Fake/{folder}/Second*/**/*");
+        }
+    }
+
+    [DataTestMethod]
+    [DataRow("")]
+    [DataRow("this is not an xml")]
+    [DataRow(@"<?xml version=""1.0"" encoding=""UTF - 8""?><AnalysisInput><Settings>")]
+    public void SonarLintXml_WithMalformedXml_DefaultBehaviour(string sonarLintXmlContent) =>
+        CheckSonarLintXmlDefaultValues(new SonarLintXmlReader(SourceText.From(sonarLintXmlContent), LanguageNames.CSharp));
+
+    [TestMethod]
+    public void SonarLintXmlSettings_MissingProperties_DefaultBehaviour() =>
```

Comment:
I suggest creating a test scenario where the settigns are really just missing and not non existent. So you can check that even if 1 or 2 are missing the rest is correctly parsed.